src/Plugin/DoctrineDBALPlugin/CloseConnections.php

```diff
@@ -32,6 +32,12 @@ public function cleanUp()
                 throw new \LogicException('Expected only instances of Connection');
             }
 
+            if ($connection->isTransactionActive() && $connection->isRollbackOnly()) {
+                $this->logger->debug('Rolling back active transaction in rollback only state', ['connection' => $name]);
+
+                $connection->rollBack();
```

Comment:
What happens if this throws?

Reply:
Thanks for taking the time to look at this @Ocramius! If would probably be smart to catch that exception, log it, and... let it go. What do you think @ruudk ?

Reply:
Having an exception bubble up is fine, this is more about general behaviour questions, eventually to be covered by tests

Reply:
Yeah @Ocramius tnx for taking the time. I agree that having an exception bubble up is fine and that we should cover this in a test.

Reply:
Great feedback. I will address this and add proper tests for it.

My biggest concern was if there would be any implications with handling the rollback inside the cleaner. But to me, that makes much sense. The cleaners are usually called after the execution of the actual code (RabbitMQ consumer job for example). The next job should always be in a ready state. If the transaction was somehow marked as rollback only, it should be reverted and cleaned up.